### ISSUE TITLE
[REVIEW]: taurenmd: A command-line interface for analysis of Molecular Dynamics simulations.

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2175]
**Submitting author:** @joaomcteixeira (<a href="http://orcid.org/0000-0002-9113-0622">João M.C. Teixeira</a>)
**Repository:** <a href="https://github.com/joaomcteixeira/taurenmd" target ="_blank">https://github.com/joaomcteixeira/taurenmd</a>
**Version:** v0.8.9
**Editor:** @richardjgowers
**Reviewer:** @amritagos, @luthaf
**Archive:** <a href="https://doi.org/10.5281/zenodo.3843326" target="_blank">10.5281/zenodo.3843326</a>

## Status

[![status](https://joss.theoj.org/papers/2b9b2dc19e4a8f38a4822486d2a9898f/status.svg)](https://joss.theoj.org/papers/2b9b2dc19e4a8f38a4822486d2a9898f)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/2b9b2dc19e4a8f38a4822486d2a9898f"><img src="https://joss.theoj.org/papers/2b9b2dc19e4a8f38a4822486d2a9898f/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/2b9b2dc19e4a8f38a4822486d2a9898f/status.svg)](https://joss.theoj.org/papers/2b9b2dc19e4a8f38a4822486d2a9898f)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/joaomcteixeira/taurenmd" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@amritagos & @luthaf, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @richardjgowers know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/joaomcteixeira/taurenmd">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@joaomcteixeira) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/joaomcteixeira/taurenmd">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@joaomcteixeira) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @amritagos, @luthaf it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1038/267585a0 is OK
- 10.1002/wcms.1449 is OK
- 10.1021/ct400341p is OK
- 10.1002/jcc.21787 is OK
-  10.25080/Majora-629e541a-00e  is OK
- 10.1016/j.bpj.2015.08.015 is OK
- 10.1002/jcc.23753 is OK
- 10.12688/f1000research.17456.1 is OK
- 10.1371/journal.pcbi.1005659 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2175 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/lib/whedon.rb:158:in `new': could not find repository from 'tmp/2175' (Rugged::RepositoryError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/lib/whedon.rb:158:in `detect_languages'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/lib/whedon.rb:93:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/lib/whedon/processor.rb:36:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/lib/whedon/processor.rb:36:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/bin/whedon:55:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-45a043c4bfc2/bin/whedon:116:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by arfon** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02175/joss.02175/10.21105.joss.02175.pdf)

**Comment by arfon** [comment]
@amritagos, @Luthaf - 
please carry out your review in this issue by updating the checklist above and giving feedback in this issue. The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html

Any questions/concerns please let @richardjgowers know.

**Comment by amritagos** [comment]
@richardjgowers @joaomcteixeira I am finished with my review.

`taurenmd` ties together some popular MD analysis libraries with an easy-to-use command-line interface. It is also possible to add new functionality, using the existing code structure. In particular, supported file formats include GROMACS trajectory files, which (to the best of my knowledge) are not supported by (OVITO)[https://www.ovito.org]. This is a neat, fairly well-documented software, which can make MD analysis accessible to newcomers to the field as well. I'm happy to recommend this for publication.

**Comment by joaomcteixeira** [comment]
Dear @amritagos 
I appreciate your comments of approval and positive feedback. I look forward to future interactions to continue developing `taurenmd` as a community-based software.
Cheers,

**Comment by joaomcteixeira** [comment]
Hi @richardjgowers and @Luthaf,

Please let me know if there are any additional materials or resources I can provide to facilitate your review process.

Best,

**Comment by richardjgowers** [comment]
Hi @joaomcteixeira we're waiting on the other review, which might be currently delayed.  I think for now we'll just have to wait a bit.

**Comment by joaomcteixeira** [comment]
Hi @richardjgowers ,
@Luthaf made some comments on the review as issues in `taurenmd` repository, I am now addressing them: https://github.com/joaomcteixeira/taurenmd/issues/35, https://github.com/joaomcteixeira/taurenmd/issues/36, https://github.com/joaomcteixeira/taurenmd/issues/41, https://github.com/joaomcteixeira/taurenmd/issues/45

**Comment by joaomcteixeira** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02175/joss.02175/10.21105.joss.02175.pdf)

**Comment by joaomcteixeira** [comment]
@richardjgowers @Luthaf 

The new article version addresses the issues referred by @Luthaf in https://github.com/joaomcteixeira/taurenmd/issues/45

I look forward to your comments, Thanks.

**Comment by Luthaf** [comment]
With the latest changes to documentation done by @joaomcteixeira, I am happy to recommend this paper for publication in JOSS!

**Comment by joaomcteixeira** [comment]
Thanks to the reviewers and the editors for all their help and insights. This review process was very fruitful, indeed! And I enjoy much this online and open process in JOSS, it really pushes scientific software development further.

**Comment by richardjgowers** [comment]
@openjournals/joss-eics this is good to go!

**Comment by arfon** [comment]
> @openjournals/joss-eics this is good to go!

Thanks @richardjgowers. In order to notify the EiC team (and to create final proofs) you need to as @whedon to `@whedon accept`. That said, it looks like this submission still needs an archive DOI for the software.

@joaomcteixeira - At this point could you make a new release of this software that includes the changes that have resulted from this review. Then, please make an archive of the software in Zenodo/figshare/other service and update this thread with the DOI of the archive? For the Zenodo/figshare archive, please make sure that:

- The title of the archive is the same as the JOSS paper title
- That the authors of the archive are the same as the JOSS paper authors

I can then move forward with accepting the submission.

**Comment by joaomcteixeira** [comment]
@arfon a quick question before doing anything, sorry for making you repeat, but I want to double-check beforehand.
* all the additions from this review are already part of the deployed `taurenmd`, currently version `v0.9.3`, which was done immediately after each PR for each new addition.
* new versions are automatically deployed in Zenodo (currently,  DOI: 10.5281/zenodo.3843326 ) -> https://zenodo.org/record/3843326#.XtQ3ORZq3ys
* latest git release: https://github.com/joaomcteixeira/taurenmd/releases

Do I still need to pack another Zenodo release?

Thank you

**Comment by arfon** [comment]
:+1: then I think this is in good shape.

**Comment by arfon** [comment]
@whedon set 10.5281/zenodo.3843326 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3843326" target="_blank">10.5281/zenodo.3843326</a> is the archive.

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1038/267585a0 is OK
- 10.1002/wcms.1449 is OK
- 10.1021/ct400341p is OK
- 10.1002/jcc.21787 is OK
-  10.25080/Majora-629e541a-00e  is OK
- 10.1016/j.bpj.2015.08.015 is OK
- 10.1002/jcc.23753 is OK
- 10.12688/f1000research.17456.1 is OK
- 10.1371/journal.pcbi.1005659 is OK
- 10.1016/j.cpc.2009.05.011 is OK
- 10.1016/j.softx.2015.06.001 is OK

MISSING DOIs

- https://doi.org/10.1016/0263-7855(96)00018-5 may be missing for title: VMD – Visual Molecular Dynamics

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1462

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1462, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@richardjgowers - ☝️ this is the point we ask you to try and reach in the review process before handing off to the EiC team.

**Comment by kyleniemeyer** [comment]
👋 that's me! (for the week)

**Comment by kyleniemeyer** [comment]
Hi @joaomcteixeira, just a few minor comments on the paper before I publish it:

- typo in the third paragraph: "To date, taurenmd does not operates in" -> "To date, taurenmd does not operate in"
- in the first paragraph of Implementation, "etc..." should just be "etc." (and not in italics
- same paragraph: "...; yet, simple, flat, easy to read, and extensible." should be something like "...; yet, it is simple, flat, easy to read, and extensible."
- please add the DOI to the Humphrey 1996 reference (10.1016/0263-7855(96)00018-5)

Once you've made these changes, please let us know here, and we'll get the article published!

**Comment by joaomcteixeira** [comment]
Hi @kyleniemeyer ,
I have corrected your highlights in the article and the new version is already merged to `master`. It compiled properly using the JOSS preview tool. Do I need to create another Zenodo DOI with this new paper revision?

**Comment by kyleniemeyer** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02175/joss.02175/10.21105.joss.02175.pdf)

**Comment by kyleniemeyer** [comment]
@joaomcteixeira thanks! No, there's no real need for another archive, unless the software itself has changed.

**Comment by kyleniemeyer** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1038/267585a0 is OK
- 10.1002/wcms.1449 is OK
- 10.1021/ct400341p is OK
- 10.1002/jcc.21787 is OK
-  10.25080/Majora-629e541a-00e  is OK
- 10.1016/j.bpj.2015.08.015 is OK
- 10.1002/jcc.23753 is OK
- 10.12688/f1000research.17456.1 is OK
- 10.1371/journal.pcbi.1005659 is OK
- 10.1016/0263-7855(96)00018-5 is OK
- 10.1016/j.cpc.2009.05.011 is OK
- 10.1016/j.softx.2015.06.001 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1463

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1463, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by joaomcteixeira** [comment]
Thanks @kyleniemeyer . No, nothing on the code has changed, just your text suggestions for the manuscript.

Best wishes,

**Comment by kyleniemeyer** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1267944039950794753) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1464
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02175](https://doi.org/10.21105/joss.02175)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by kyleniemeyer** [comment]
@joaomcteixeira congratulations on your article's publication in JOSS!

Many thanks to @amritagos and @Luthaf for reviewing this submission, and @richardjgowers for editing.

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02175/status.svg)](https://doi.org/10.21105/joss.02175)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02175">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02175/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02175/status.svg
   :target: https://doi.org/10.21105/joss.02175
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02175/status.svg)](https://doi.org/10.21105/joss.02175)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss

**Comment by joaomcteixeira** [comment]
Thanks @kyleniemeyer @richardjgowers @Luthaf @amritagos for all your work and help on this publication! Cheers :clinking_glasses:

**Comment by joaomcteixeira** [comment]
Dear @kyleniemeyer and @arfon,
I just noticed that in the article citation string (web, PDF bottom of pages, bibtex), as well as, in the article JOSS page, my name appears incomplete as 'Teixeira, J.M.' or 'João Teixeira', instead of my scientific full name: `Teixeira, J.M.C.` or `João M.C. Teixeira` (as in the main article authors section).

Is it possible to update these citation references to my full scientific name?

Thank you in advance.

**Comment by arfon** [comment]
Thanks for flagging this @joaomcteixeira. This should now be fixed. Please note the PDF on the JOSS site will take some time to update as it is cached.

**Comment by joaomcteixeira** [comment]
@arfon thank you very much, it looks okay now, also in the PDF file. I am happy to see that this resulted in an actual PR to improve the authors experience in JOSS.
Best wishes,
